app/src/main/java/com/example/cleanarchitecture/activity/MainActivity.kt

```diff
@@ -25,22 +26,36 @@ class MainActivity : AppCompatActivity() {
 
         viewModel.liveDataState.observe(this, { updateUI(it) })
 
-        binding.buttonMainActivityIdFetch.setOnClickListener { viewModel.onFetchCharactersClicked() }
+        viewModel.fetchCharacters()
     }
 
-    private fun updateUI(characterData: Event<Data<CharacterDataWrapperEntity>>?) {
-        when (characterData?.peekContent()?.responseType) {
+    private fun updateUI(characterData: Event<Data<CharacterDataWrapperEntity>>) {
+        when (characterData.peekContent().responseType) {
```

Comment:
please don't force getting the content, if you do that the event class is useless

Reply:
done